src/client/dfuse/dfuse_main.c

```diff
@@ -364,6 +371,26 @@ main(int argc, char **argv)
 		D_GOTO(out_debug, ret = -DER_NO_HDL);
 	}
 
+	if (dfuse_info->di_threaded && !have_thread_count) {
+		cpu_set_t cpuset;
+
+		rc = sched_getaffinity(0, sizeof(cpuset), &cpuset);
+		if (rc != 0) {
+			printf("Failed to get cpuset information\n");
+			exit(1);
+		}
+
+		dfuse_info->di_thread_count = CPU_COUNT(&cpuset);
+
+		/* Reserve one CPU thread for the daos event queue */
+		dfuse_info->di_thread_count -= 1;
+
+		if (dfuse_info->di_thread_count < 1) {
```

Comment:
I think we need a similar check for the passed in `--thread-count`. Right now, dfuse lets me pass in 0 threads.
Maybe check this outside of this if statement to account for the passed in value or cpuset?

Reply:
I'll fix this in https://github.com/daos-stack/daos/pull/3894 which is changing the command line processing anyway.